crates/hyperswitch_connectors/src/connectors/elavon.rs

```diff
@@ -33,20 +34,51 @@ use hyperswitch_interfaces::{
     types::{self, Response},
     webhooks,
 };
-use masking::{ExposeInterface, Mask};
+use masking::{Secret, WithoutType};
+use quick_xml::de::from_str;
+use serde::{de::DeserializeOwned, Serialize};
 use transformers as elavon;
 
 use crate::{constants::headers, types::ResponseRouterData, utils};
 
+pub fn parse_struct<T: DeserializeOwned>(xml_data: &[u8]) -> Result<T, errors::ConnectorError> {
+    let response_str = str::from_utf8(xml_data)
+        .map_err(|e| {
+            router_env::logger::error!("Error converting response data to UTF-8: {:?}", e);
+            errors::ConnectorError::ResponseDeserializationFailed
+        })?
+        .trim();
+    let result: T = from_str(response_str).map_err(|e| {
+        router_env::logger::error!("Error deserializing XML response: {:?}", e);
+        errors::ConnectorError::ResponseDeserializationFailed
+    })?;
+
+    Ok(result)
+}
+pub fn struct_to_xml<T: Serialize>(
+    item: &T,
+) -> Result<HashMap<String, Secret<String, WithoutType>>, errors::ConnectorError> {
+    let xml_content = quick_xml::se::to_string_with_root("txn", &item).map_err(|e| {
+        router_env::logger::error!("Error serializing Struct: {:?}", e);
+        errors::ConnectorError::ResponseDeserializationFailed
+    })?;
+
+    let mut result = HashMap::new();
+    result.insert(
+        "xmldata".to_string(),
+        Secret::<_, WithoutType>::new(xml_content),
+    );
+    Ok(result)
+}
```

Comment:
Can we move this to connector utils ?

Reply:
moved `parse_struct` to the utils, but the `struct_to_xml` contains some steps specific to Paybox (such as adding the root item and "xmldata".to_string())